src/integer/mat_z/properties.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright © 2023 Phil Milewski
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes functionality about properties of [`MatZ`] instances.
+
+use super::MatZ;
+use flint_sys::fmpz_mat::{fmpz_mat_is_one, fmpz_mat_is_square, fmpz_mat_is_zero};
+
+impl MatZ {
+    /// Checks if a [`MatZ`] is the identity matrix.
+    ///
+    /// Returns true if every diagonal entry of the upper square matrix is `1`
+    /// and every other entry is `0`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let value = MatZ::from_str("[[1, 0],[0, 1]]").unwrap();
+    /// assert!(value.is_identity())
+    /// ```
+    pub fn is_identity(&self) -> bool {
+        1 == unsafe { fmpz_mat_is_one(&self.matrix) }
+    }
+
+    /// Checks if a [`MatZ`] is a square matrix.
+    ///
+    /// Returns `true` if the number of rows and columns is identical.
```

Comment:
It would be nice, if you could stay consistent in the description of these functions accross types